_posts/2019-09-09-Shake-Up.md

```diff
@@ -25,15 +25,15 @@ The no-modification rule is apparent from the text, to those vanishingly few sou
 >
 > Notwithstanding any other provision of this License, <mark>if you modify the Program</mark>, your modified version must prominently offer all users interacting with it remotely through a computer network ... an opportunity to receive the Corresponding Source of your version ...
 
-The ability to "build laywers between the AGPL 3 code and proprietary code" was less well known, not evident from a passive top-to-bottom skim, and rooted in nook, crannies, and interactions of various defined terms and sub-rules.  I've described an "API Loophole" and a "Container Loophole" in [earlier writing](https://writing.kemitchell.com/2018/11/04/Copyleft-Bust-Up.html).
+The ability to "build layers between the AGPL 3 code and proprietary code" was less well known, not evident from a passive top-to-bottom skim, and rooted in nook, crannies, and interactions of various defined terms and sub-rules.  I've described an "API Loophole" and a "Container Loophole" in [earlier writing](https://writing.kemitchell.com/2018/11/04/Copyleft-Bust-Up.html).
 
 MongoDB's representatives also alluded to these issues in their written rationales for their Server Side Public License, but seemed to consciously avoid saying too much specifically.  I'd guess they didn't want to give credence to the arguments, which could never be perfectly airtight.  And it wouldn't help them to spread knowledge, given many previous versions of Mongo are out there under AGPL, and they can't take those grants back.
 
 ## Not Discriminating
 
 Kate gives the OSI credit for standing hard on a principle of no discrimination in licenses, but connects their failure to anoint any new license meeting companies' needs to ensuing license proliferation and shifts toward more proprietary code.
 
-I catch myself falling back on an expeident, feel-good notion of "nondiscrimination", too.  But it's not consistent with open source history.  If "open source" includes the preexisting free software copyleft licenses, and not just as payment of old political debt or for stepping stones to march over, then open source included explicitly discriminatory licenses.  RMS wrote GPL to discrimiante against propreitary takers.  If some sofwtare companies used the GPLs to poke other software companies in the eye for closing soruce code, so much the better.
+I catch myself falling back on an expedient, feel-good notion of "nondiscrimination", too.  But it's not consistent with open source history.  If "open source" includes the preexisting free software copyleft licenses, and not just as payment of old political debt or for stepping stones to march over, then open source included explicitly discriminatory licenses.  RMS wrote GPL to discriminate against proprietary takers.  If some software companies used the GPLs to poke other software companies in the eye for closing source code, so much the better.
```

Comment:
There were a number of typos in here, and Github won't display the individual changes unless you use the "rich diff" button. :-/